client/src/components/table/CreateTable.tsx

```diff
@@ -0,0 +1,237 @@
+// Aby utworzyć nową tabelę należy podać jej dane oraz opcje kolumn
+//   titleName: string;
+//   keyName: string;
+//   sortType?: 'number' | 'string' | 'date';
+//   ----- typ sortowania, jeśli nie podamy to nie będzie sortowania
+//   search?: boolean;
+//   ----- czy kolumna ma mieć możliwość wyszukiwania
+//   actions?: actions[];
+//   ----- opcje akcji, jeśli nie podamy to nie będzie akcji
+
+import { Space, Table, Button, Input } from 'antd';
+import type { InputRef } from 'antd';
+import type { ColumnsType, ColumnType } from 'antd/es/table';
+import type { FilterConfirmProps } from 'antd/es/table/interface';
+import { SearchOutlined } from '@ant-design/icons';
+
+import type { TableColumns, DataType } from '@/utils';
+
+import React, { useRef, useState } from 'react';
+import Highlighter from 'react-highlight-words';
+
+interface TableProps {
+  data: DataType[];
+  columnsOptions: TableColumns[];
+}
+
+interface SearchState {
+  searchText: string;
+  searchedColumn: string;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  searchInput: any;
+  setSearchText: (value: string) => void;
+  setSearchedColumn: (value: string) => void;
+}
+
+const CreateColumns = (columnsOptions: TableColumns[], searchState: SearchState) => {
```

Comment:
Would be a nice idea to use generics - so we aren't always bound to single DataType interface